documentation/modules/kafkaconnectors/proc-deploying-kafkaconnector.adoc

```diff
@@ -0,0 +1,57 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-connect.adoc
+
+[id='proc-deploying-kafkaconnector-{context}']
+= Deploying a `KafkaConnector` resource to Kafka Connect
+
+Deploy the example `KafkaConnector` to a Kafka Connect cluster. The example YAML file is configured to create a `FileStreamSourceConnector` to send each line of the Kafka license file as a message in Kafka.
+
+.Prerequisites
+
+* A Kafka Connect deployment in which xref:proc-enabling-kafkaconnectors-deployment-configuration-kafka-connect[`KafkaConnectors` are enabled]
+* A running Cluster Operator
+
+.Procedure
+
+. Edit the `examples/connector/source-connector.yaml` file:
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: `KafkaConnector`
+metadata:
+  name: my-source-connector <1>
+  labels:   
+    strimzi.io/cluster: my-connect-cluster <2>
+spec:
+  class: org.apache.kafka.connect.file.FileStreamSourceConnector <3>
+  tasksMax: 2 <4>
+  config: <5>
+    file: "/opt/kafka/LICENSE"
+    topic: my-topic
```

Comment:
Can we add also a simple Sink connector example?

Reply:
Sure. Is there somewhere I can borrow one from?

Reply:
@scholzj the example given comes from the `examples` directory. The source connector example is weak enough (reading the licence into a topic) and I didn't do a sink connector example because that would be even more weird – let's just go filling the connect filesystem with some arbitrary data read from a topic. We're basically teaching users stuff which they shouldn't be doing.

I wonder if we should provide your echo connectors in https://github.com/strimzi/client-examples. That way at least the examples wouldn't be so confusing for users.

Reply:
@tombentley Well, I guess to make it work in the examples, we would need to bundle it with our Kafka images. I don't have problem with it, if we want to do that, I can get it pushed to Maven Central and we can pull it in 3rd party libs. Alternatively, we could move it under Strimzi, but I would probably prefer separate repo to make it easy to push it to Maven.

Reply:
Whatever works easiest suits me.